File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/controller/MessageRestController.java
package com.raxim.myscoutee.profile.controller;

import org.springframework.data.rest.webmvc.RepositoryRestController;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestHeader;

import com.raxim.myscoutee.profile.data.document.mongo.Message;
import com.raxim.myscoutee.profile.service.MessageService;

@RepositoryRestController("messages")
public class MessageRestController {

    private final MessageService messageService;

    public MessageRestController(MessageService messageService) {
        this.messageService = messageService;
    }

    @GetMapping("")
    public ResponseEntity<Message> getMsg(@RequestHeader("accept-language") String locale) {
        Message message = messageService.getMessageByLang(locale);
        return ResponseEntity.ok(message);
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/controller/MessageRestController.kt
package com.raxim.myscoutee.profile.controller

import com.raxim.myscoutee.profile.data.document.mongo.Message
import com.raxim.myscoutee.profile.service.MessageService
import org.springframework.data.rest.webmvc.RepositoryRestController
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping

@RepositoryRestController
@RequestMapping("messages")
class MessageRestController(
    private val messageService: MessageService
) {

    @GetMapping("")
    fun getMsg(
        @RequestHeader("accept-language") locale: String
    ): ResponseEntity<Message> {
        val message = this.messageService.getMessageByLang(locale);
        return ResponseEntity.ok(message!!)
    }

}